hack/boilerplate/boilerplate.py

```diff
@@ -16,22 +16,54 @@
 
 from __future__ import print_function
 
+import argparse
 import json
 import mmap
 import os
 import re
 import sys
 
-def PrintError(*err):
-  print(*err, file=sys.stderr)
+parser = argparse.ArgumentParser()
+parser.add_argument("filenames", help="list of files to check, all files if unspecified", nargs='*')
+args = parser.parse_args()
+
+rootdir = os.path.dirname(__file__) + "/../../"
+rootdir = os.path.abspath(rootdir)
+
+def get_refs():
+    extRE = re.compile("boilerplate.([^.]+).txt")
+
+    refs = {}
+    for root, dirs, files in os.walk(os.path.join(rootdir, "hack/boilerplate/")):
+        for pathname in files:
+            match = extRE.match(pathname)
+            if not match:
+                continue
+            extension = match.group(1)
```

Comment:
An alternate way to do this would be
```python
for path in glob.glob(os.path.join(rootdir, "hack/boilerplate/boilerplate.*.txt")):
  extension = os.path.basename(path).split(".")[1]
  ...
```
This would avoid the walk when you really don't want to do a recursive traversal.